clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -177,6 +177,22 @@ check_branch_target_after_branch()
         if (!run_checker(memrefs, false))
             return false;
     }
+    // Invariant should not trigger when it detects a timestamp at the start of a new
+    // chunk.
+    {
+        std::vector<memref_t> memrefs = {
+            gen_instr(1, 1),
+            gen_branch(1, 2),
+            gen_marker(1, TRACE_MARKER_TYPE_CHUNK_INSTR_COUNT, 1),
+            gen_marker(1, TRACE_MARKER_TYPE_CHUNK_FOOTER, 3),
+            gen_marker(1, TRACE_MARKER_TYPE_TIMESTAMP, 3),
```

Comment:
Per the prior comment on reader_t hiding this probably we should only have the footer (which I assume still triggers the false pos).